src/modules/Game.class.js

```diff
@@ -1,68 +1,240 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
-class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
+export default class Game {
   constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.initialState = initialState || [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+    this.state = structuredClone(this.initialState);
+    this.score = 0;
+    this.status = 'idle';
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
-}
+  moveLeft() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    const prevState = structuredClone(this.state);
+
+    for (let i = 0; i < 4; i++) {
+      const row = this.state[i].filter((cell) => cell !== 0);
+
+      for (let j = 0; j < row.length - 1; j++) {
+        if (row[j] === row[j + 1]) {
+          row[j] *= 2;
+          this.score += row[j];
+          row.splice(j + 1, 1);
+        }
+      }
+
+      while (row.length < 4) {
+        row.push(0);
+      }
+      this.state[i] = row;
+    }
+
+    if (prevState !== structuredClone(this.state)) {
+      this.addRandomTile();
+      this.updateGameStatus();
+    }
+  }
+
+  moveRight() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    const prevState = structuredClone(this.state);
+
+    for (let i = 0; i < 4; i++) {
+      const row = this.state[i].filter((cell) => cell !== 0);
+
+      for (let j = row.length - 1; j > 0; j--) {
+        if (row[j] === row[j - 1]) {
+          row[j] *= 2;
+          this.score += row[j];
+          row.splice(j - 1, 1);
+          j--;
+        }
+      }
+
+      while (row.length < 4) {
+        row.unshift(0);
+      }
+      this.state[i] = row;
+    }
+
+    if (prevState !== structuredClone(this.state)) {
+      this.addRandomTile();
+      this.updateGameStatus();
+    }
```

Comment:
This logic is similar in every method, try to put it in a helper function and use it everywhere